File: Napakalaki/src/napakalaki/PruebaNapakalaki.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;

import java.lang.reflect.Array;
import java.util.ArrayList;
import java.util.Arrays;

/**
 *
 * @author chen__000
 */
public class PruebaNapakalaki {

   static void filtro_niveles10( ArrayList<Monster> Array){
        for(int i=0;i<Array.size();i++){
             if(Array.get(i).getCombatLevel()>10)
                 System.out.println(Array.get(i));
                  System.out.println(" ");
        }
    }
   
   static void filtro_bc_pierde_nivel(ArrayList<Monster> Array){
       for(int i=0;i<Array.size();i++){
          if(Array.get(i).bc.getSpecificHiddenTreasures().size()==0&&Array.get(i).bc.getSpecificVisibleTreasures().size()==0&&Array.get(i).bc.getnHiddenTreasures()==0 && Array.get(i).bc.getnVisibleTreasures()==0 && Array.get(i).bc.getLevels()!=0 ){
                  System.out.println(Array.get(i));
                  System.out.println(" ");
           }
       }
   }
   
   
   static void filtro_gana_nivel(ArrayList<Monster> Array){
       for(int i=0;i<Array.size();i++){
          if(Array.get(i).price.getLevel()>1){
              System.out.println(Array.get(i));
              System.out.println(" ");
          }
       }
   }
   
      static void filtro_bc_pierde_tesoros(ArrayList<Monster> Array){
       for(int i=0;i<Array.size();i++){
          if(Array.get(i).bc.getSpecificHiddenTreasures().size()!=0 || Array.get(i).bc.getSpecificVisibleTreasures().size()!=0||Array.get(i).bc.getnHiddenTreasures()!=0 || Array.get(i).bc.getnVisibleTreasures()!=0  ){
                  System.out.println(Array.get(i));
                  System.out.println(" ");
           }
       }
   }
  
    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        // TODO code application logic here
        ArrayList<Monster> monstruos = new ArrayList();
        
        //creacion de monstruos
        
        BadConsequence badConsequence1 = new BadConsequence("Pierdes 5 niveles y 3 tesoros visibles",5,3,0);
        Prize prize1 = new Prize(4,2);
        monstruos.add(new Monster("El rey de rosa",13,badConsequence1,prize1));
        
        BadConsequence badConsequence2 = new BadConsequence("Te atrapan para llevarte de fiesta y te dejan caer en mitad del vuelo. Descarta 1 mano visible y 1 mano oculta",0,
                                                             new ArrayList(Arrays.asList(TreasureKind.ONEHAND)),
                                                             new ArrayList(Arrays.asList(TreasureKind.ONEHAND)));
        Prize prize2 = new Prize(4,1);
        monstruos.add(new Monster("Angeles de la noche ibicenca",14,badConsequence2,prize2));
        
        BadConsequence badConsequence3 = new BadConsequence("Pierdes tu armadura visible y otra oculta",0,
                                                             new ArrayList(Arrays.asList(TreasureKind.ARMOR)),
                                                             new ArrayList(Arrays.asList(TreasureKind.ARMOR)));
        Prize prize3 = new Prize(2,1);
        monstruos.add(new Monster("3 Byakhees de bonanza",8,badConsequence3,prize3));
    
        BadConsequence badConsequence4 = new BadConsequence ("Embobados con el lindo primigenio te descarta de tu casco visible",0
                                                                ,new ArrayList(Arrays.asList(TreasureKind.HELMET))
                                                                ,new ArrayList(Arrays.asList()));
        Prize prize4 = new Prize(1,1);
        monstruos.add(new Monster("Chibithulhu",2,badConsequence4,prize4));
    
       BadConsequence badConsequence5 = new BadConsequence ("El primordial bostezo contagioso, pierdes el calzado visible",0
                                                                ,new ArrayList(Arrays.asList(TreasureKind.SHOES))
                                                                ,new ArrayList(Arrays.asList()));
        Prize prize5 = new Prize(1,1);
        monstruos.add(new Monster("El sopor de Dunwich",2,badConsequence5,prize5));
    
       BadConsequence badConsequence6 = new BadConsequence ("Pierdes todos los tesoros visibles",0
                                                                ,new ArrayList(Arrays.asList(TreasureKind.ARMOR,TreasureKind.BOTHHANDS,TreasureKind.HELMET,TreasureKind.ONEHAND,TreasureKind.SHOES))
                                                                ,new ArrayList(Arrays.asList()));
        Prize prize6 = new Prize(3,1);
        monstruos.add(new Monster("El gorron en el umbral",10,badConsequence6,prize6));
    
        BadConsequence badConsequence7 = new BadConsequence ("Pierdes la armadura visible",0
                                                                ,new ArrayList(Arrays.asList(TreasureKind.ARMOR))
                                                                ,new ArrayList(Arrays.asList()));
        Prize prize7 = new Prize(2,1);
        monstruos.add(new Monster("H.P. Munchcraft",6,badConsequence7,prize7));
    
        BadConsequence badConsequence8 = new BadConsequence ("Sientes bichos bajo la ropa. Descarta la armadura visible",0
                                                                ,new ArrayList(Arrays.asList(TreasureKind.ARMOR))
                                                                ,new ArrayList(Arrays.asList()));
        Prize prize8 = new Prize(1,1);
        monstruos.add(new Monster("Bichgooth",2,badConsequence8,prize8));
    
        BadConsequence badConsequence9 = new BadConsequence("Toses los pulmones y pierdes 2 niveles",2,0,0);
        Prize prize9 = new Prize(1,1);
        monstruos.add(new Monster("La que redacta en las tinieblas",2,badConsequence9,prize9));
        
        BadConsequence badConsequence10 = new BadConsequence("Estos monstruos resultan bastante superficiales y te aburren mortalmente. Estas muerto",true);
        Prize prize10 = new Prize(2,1);
        monstruos.add(new Monster("Los Hondos",8,badConsequence10,prize10));
        
        BadConsequence badConsequence11 = new BadConsequence("Pierdes 2 niveles y 2 tesoros ocultos",2,0,2);
        Prize prize11 = new Prize(2,1);
        monstruos.add(new Monster("Semillas Cthulhu",4,badConsequence11,prize11));
        
        BadConsequence badConsequence12 = new BadConsequence ("Te intentas escaquear. Pierdes una mano visible",0
                                                                ,new ArrayList(Arrays.asList(TreasureKind.ONEHAND))
                                                                ,new ArrayList(Arrays.asList()));
        Prize prize12 = new Prize(2,1);
        monstruos.add(new Monster("Dameargo",1,badConsequence12,prize12));
        
        BadConsequence badConsequence13 = new BadConsequence("Da mucho asquito. Pierdes 3 niveles",3,0,0);
        Prize prize13 = new Prize(1,1);
        monstruos.add(new Monster("Pollipolipo volante",3,badConsequence13,prize13));
        
        BadConsequence badConsequence14 = new BadConsequence("No le hace gracia que pronuncien mal su nombre. Estas muerto",true);
        Prize prize14 = new Prize(3,1);
        monstruos.add(new Monster("Yskhtihyssg-Goth",12,badConsequence14,prize14));
        
        BadConsequence badConsequence15 = new BadConsequence("La familia te atrapa. Estas muerto",true);
        Prize prize15 = new Prize(4,1);
        monstruos.add(new Monster("Familia feliz",1,badConsequence15,prize15));
        
       BadConsequence badConsequence16 = new BadConsequence ("La quinta directiva primaria te obliga a perder 2 niveles y un tesoro 2 manos visible",2
                                                                ,new ArrayList(Arrays.asList(TreasureKind.BOTHHANDS))
                                                                ,new ArrayList(Arrays.asList()));
        Prize prize16 = new Prize(2,1);
        monstruos.add(new Monster("Roboggoth",8,badConsequence16,prize16));
        
        BadConsequence badConsequence17 = new BadConsequence ("Te asusta en la noche. Pierdes un casco visible",0
                                                                ,new ArrayList(Arrays.asList(TreasureKind.HELMET))
                                                                ,new ArrayList(Arrays.asList()));
        Prize prize17 = new Prize(1,1);
        monstruos.add(new Monster("El espia",5,badConsequence17,prize17));
        
        BadConsequence badConsequence18 = new BadConsequence("Menudo susto te llevas. Pierdes 2 niveles y 5 tesoros visibles",2,5,0);
        Prize prize18 = new Prize(1,1);
        monstruos.add(new Monster("El Lenguas",20,badConsequence18,prize18));
        
        BadConsequence badConsequence19 = new BadConsequence ("Te faltan manos para tanta cabeza. Pierdes 3 niveles y tus tesoros visibles de las manos",3
                                                                ,new ArrayList(Arrays.asList(TreasureKind.BOTHHANDS))
                                                                ,new ArrayList(Arrays.asList()));
        Prize prize19 = new Prize(1,1);
        monstruos.add(new Monster("Bicefalo",20,badConsequence19,prize19));
        

        
        //comprobaciones

//        Monster m1;
//        BadConsequence badConsequence20 = new BadConsequence("Te atrapan para llevarte de fiesta y te dejan caer en mitad del vuelo. Descarta 1 mano visible y 1 mano oculta",0,
//                                                             new ArrayList(Arrays.asList(TreasureKind.ONEHAND)),
//                                                             new ArrayList(Arrays.asList(TreasureKind.ONEHAND)));
//        Prize prize20 = new Prize(4,1);
//        m1=new Monster("Angeles de la noche ibicenca",14,badConsequence20,prize20);
//        System.out.println(m1);
//        
        
        //filtro_niveles10(monstruos);
        
         //filtro_bc_pierde_nivel(monstruos);

      // filtro_gana_nivel(monstruos);
        
      // filtro_bc_pierde_tesoros(monstruos);
       
     
  
    }
        
}

File: Napakalakiruby/lib/PruebaNapakalaki.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative "prize.rb"
require_relative "bad_consequence.rb"
require_relative "monster.rb"
require_relative "treasure_kind.rb"
require_relative "player.rb"
require_relative "treasure.rb"

monsters=Array.new

def filtro_nivel_10(monsters)
  i=0
while (i<monsters.size)
  if(monsters[i].getCombatLevel>10)
    puts monsters[i] 
    puts "\n"
  end
  i=i+1
end
end

def filtro_gana_nivel(monsters)
   i=0
while (i<monsters.size)
  if(monsters[i].getPrice.getLevel>1)
    puts monsters[i] 
    puts "\n"
  end
  i=i+1
end
end

def fitro_bc_pierdenivel(monsters)
     i=0
while (i<monsters.size)
  if(monsters[i].getBadConsequence.getLevels!=0&&monsters[i].getBadConsequence.getnVisibleTreasures==0&&monsters[i].getBadConsequence.getnHiddenTreasures==0&&monsters[i].getBadConsequence.getnSpecificVisibleTreasures==0&&monsters[i].getBadConsequence.getnSpecificHiddenTreasures==0 )
    puts monsters[i] 
    puts "\n"
  end
  i=i+1
end
end

def fitro_bc_pierdetesoros(monsters)
     i=0
while (i<monsters.size)
  if((monsters[i].getBadConsequence.getnVisibleTreasures!=0&&monsters[i].getBadConsequence.getDeath==false) || (monsters[i].getBadConsequence.getnHiddenTreasures!=0&&monsters[i].getBadConsequence.getDeath==false) || (monsters[i].getBadConsequence.getnSpecificVisibleTreasures!=0&&monsters[i].getBadConsequence.getDeath==false) || (monsters[i].getBadConsequence.getnSpecificHiddenTreasures!=0&&monsters[i].getBadConsequence.getDeath==false) )
    puts monsters[i] 
    puts "\n"
  end
  i=i+1
end
end

      #creacion de los monstruos

        prize1 = Prize.new(4,2)
        badConsequence1 = BadConsequence.newLevelNumberOfTreasures("Pierdes 5 niveles y 3 tesoros visibles",5,3,0);
        monsters << Monster.new("el rey de rosa",13,prize1,badConsequence1)
        
        prize2 = Prize.new(4,1)
        badConsequence2 = BadConsequence.newLevelNumberOfTreasures("Te atrapan para llevarte de fiesta y te dejan caer en mitad del vuelo. Descarta 1 mano visible y 1 mano oculta",0,[TreasureKind::ONEHAND],[TreasureKind::ONEHAND])
        monsters<<Monster.new("Angeles de la noche ibicena",14,prize2,badConsequence2)
        
        prize3 = Prize.new(2,1)
        badConsequence3 = BadConsequence.newLevelNumberOfTreasures("Pierdes tu armadura visible y otra oculta",0,[TreasureKind::ARMOR],[TreasureKind::ARMOR])
        monsters<<Monster.new("3 Byakhees de bonanza",8,prize3,badConsequence3)
        
        prize4=Prize.new(1,1)
        badConsequence4=BadConsequence.newLevelNumberOfTreasures("Embobados con el lindo primigenio te descartas de tu casco visible",0,[TreasureKind::HELMET],[])
        monsters<<Monster.new("Chibithulhu",2,prize4,badConsequence4)
        
        prize5=Prize.new(1,1)
        badConsequence5=BadConsequence.newLevelNumberOfTreasures("El primordial bostezo contagioso. Pierdes el calzado visible",0,[TreasureKind::SHOES],[])
        monsters<<Monster.new("El sopor de Dunwich",2,prize5,badConsequence5)
        
        prize6=Prize.new(3,1)
        badConsequence6=BadConsequence.newLevelNumberOfTreasures("Pierdes todos tus tesoros visibles",0,[TreasureKind::ARMOR,TreasureKind::HELMET,TreasureKind::SHOES,TreasureKind::ONEHAND,TreasureKind::BOTHHAND],[])
        monsters<<Monster.new("H.P. Munchcraft",6,prize6,badConsequence6)
        
        prize7=Prize.new(1,1)
        badConsequence7=BadConsequence.newLevelNumberOfTreasures("Sientes bichos bajo la ropa. Descarta la armadura visible",0,[TreasureKind::ARMOR],[])
        monsters<<Monster.new("Bichgooth",2,prize7,badConsequence7)
        
        prize8=Prize.new(1,1)
        badConsequence8= BadConsequence.newLevelNumberOfTreasures("Toses los pulmones y pierdes 2 niveles",2,0,0);
        monsters << Monster.new("La que redacta en las tinieblas",2,prize8,badConsequence8)
        
        prize9=Prize.new(2,1)
        badConsequence9=BadConsequence.newDeath("Estos monstruos resultan bastante superficiales y te aburren mortalmente. Estas muerto")
        monsters << Monster.new("Los hondos",8,prize9,badConsequence9)
        
        prize10=Prize.new(2,1)
        badConsequence10= BadConsequence.newLevelNumberOfTreasures("Pierdes 2 niveles y 2 tesoros ocultos",2,0,2);
        monsters << Monster.new("Semillas Cthulhu",4,prize10,badConsequence10)
        
        prize11=Prize.new(2,1)
        badConsequence11=BadConsequence.newLevelNumberOfTreasures("Te intentas escaquear. Pierdes una mano visible",0,[TreasureKind::ONEHAND],[])
        monsters<<Monster.new("Dameargo",1,prize11,badConsequence11)
        
        prize12=Prize.new(1,1)
        badConsequence12=BadConsequence.newLevelNumberOfTreasures("Da mucho asquito. Pierdes 3 niveles",3,0,0)
        monsters<<Monster.new("Pollipolipo volante",3,prize12,badConsequence12)
        
        prize13=Prize.new(3,1)
        badConsequence13=BadConsequence.newDeath("No le hace gracia que pronuncien mal su nombre. Estas muerto")
        monsters << Monster.new("Yskhtihyssg-Goth",12,prize13,badConsequence13)
        
        prize14=Prize.new(4,1)
        badConsequence14=BadConsequence.newDeath("La familia te atrapa. Estas muerto")
        monsters << Monster.new("Familia feliz",1,prize14,badConsequence14)
        
        prize15=Prize.new(2,1)
        badConsequence15=BadConsequence.newLevelNumberOfTreasures("La quinta directiva primaria te obliga a perder 2 niveles y un tesoro 2 manos visibles",2,[TreasureKind::BOTHHAND],[])
        monsters<<Monster.new("Roboggoth",8,prize15,badConsequence15)
        
        prize16=Prize.new(1,1)
        badConsequence16=BadConsequence.newLevelNumberOfTreasures("Te asusta en la noche. Pierdes un casco visible",0,[TreasureKind::HELMET],[])
        monsters<<Monster.new("El espia",5,prize16,badConsequence16)
        
        prize17=Prize.new(1,1)
        badConsequence17= BadConsequence.newLevelNumberOfTreasures("Menudo susto te llevas. Pierdes 2 niveles y 5 tesoros visibles",2,5,0);
        monsters << Monster.new("El Lenguas",20,prize17,badConsequence17)
        
        prize18=Prize.new(1,1)
        badConsequence18=BadConsequence.newLevelNumberOfTreasures("Te faltan manos para tanta cabeza. Pierdes 3 niveles y tus tesoros visibles de las manos",3,[TreasureKind::BOTHHAND],[])
        monsters<<Monster.new("Bicefalo",20,prize18,badConsequence18)
        

#comprobacion

# prize17=Prize.new(1,1)
#        badConsequence17= BadConsequence.newLevelNumberOfTreasures("Menudo susto te llevas. Pierdes 2 niveles y 5 tesoros visibles",2,5,0);
#      
#m1=  Monster.new("El Lenguas",20,prize17,badConsequence17)
#puts m1
#



#filtro_nivel_10(monsters)

#fitro_bc_pierdenivel(monsters)

#filtro_gana_nivel(monsters)

#fitro_bc_pierdetesoros(monsters)
